new/rfcXXX-client-compat-metadata.md

```diff
@@ -0,0 +1,29 @@
+---
+RFC: unassigned
+Author: Jon Cowie (jcowie@etsy.com)
+Status: Draft
+Type: Standards Track
+---
+
+
+# Add client_compat field to cookbook metadata
+
+Increasingly, as momentum on Chef development continues to grow, it is necessary to release modifications to the recipe DSL or core resources which break backwards compatibility and result in cookbooks potentially breaking under old versions of chef-client. Further, it is often necessary to make these changes in minor chef-client releases rather than waiting multiple months for the next major release cycle. There is currently no way to surface this version requirement to Chef users.
```

Comment:
So as far as wording goes I still strongly object to this being described as introducing resources which "break backwards compatibility."  That is not what we're talking about.  In core chef these are all minor releases with *STRICTLY* backwards compatible changes in them.  As those changes are adopted by *COOKBOOKS* it produces backwards-incompatible cookbook code which cannot be used on prior versions of chef-client.